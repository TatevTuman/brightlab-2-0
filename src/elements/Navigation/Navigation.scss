@import 'static/assets/styles/mixins';

.navigation {
  display: flex;
  align-items: center;

  &-list {
    display: flex;
    align-items: center;
    padding: 0;
    margin: 0 -20px;

    &-item {
      position: relative;
      display: flex;
      align-items: center;
      margin: 0 20px;
      list-style: none;
      font-size: toRem(16);
      font-weight: 700;
      cursor: pointer;
      opacity: 0.6;
      transition: opacity 0.2s ease;
      will-change: opacity;

      &-icon {
        margin-right: toRem(20);
      }

      &-link {
        color: var(--primary)
      }

      &:after {
        position: absolute;
        left: 0;
        bottom: -#{toRem(31)};
        content: '';
        height: 100%;
        width: 100%;
        border-bottom: solid 2px var(--purple);
        transform: scaleX(0.0001);
        transition: transform 0.2s ease;
      }

      &[data-active="true"] {
        opacity: 1;

        .navigation-list-item-icon {
          //__schedule
          //__people
          //__incoming
          //__meeting-room
        }

        &:after {
          transform: scaleX(1) !important;
        }
      }

      &:hover {
        opacity: 1;

        .navigation-list-item-icon {
          //filter: invert(14%) sepia(20%) saturate(5001%) hue-rotate(217deg) brightness(103%) contrast(104%);
        }

        &:after {
          transform: scaleX(0.8);
        }
      }
    }
  }
}
