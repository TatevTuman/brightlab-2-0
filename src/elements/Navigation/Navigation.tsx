import { navigate } from 'gatsby'
import React from 'react'
import { useMatch } from '@reach/router'
import {Icon, IconName} from '@elements'
import { NavigationType, NavigationItemType } from '@types'
import './Navigation.scss'

export interface NavigationItemProps extends NavigationItemType {
  activeMatch?: boolean
}

const NavigationItem: React.FC<NavigationItemProps> = props => {
  const { path, label, icon } = props
  const match = useMatch(path)

  const isUriMatched = !!match

  return (
    <li className={'navigation-list-item'} key={path} data-active={isUriMatched} onClick={() => navigate(path)}>
      <div className={'navigation-list-item-icon'}>
        <Icon name={icon as IconName} />
      </div>
      <div className={'navigation-list-item-link'}>{label}</div>
    </li>
  )
}

export interface NavigationProps {
  navigation: NavigationType
  direction?: string
  activeMatch?: boolean
}

const Navigation: React.FC<NavigationProps> = props => {
  const { navigation, direction, activeMatch } = props

  return (
    <nav className={'navigation'} role={'navigation'} data-direction={direction}>
      <ul className={'navigation-list'}>
        {navigation.map((item, index) => {
          return <NavigationItem key={item.label + index} {...item} activeMatch={activeMatch} />
        })}
      </ul>
    </nav>
  )
}

export default Navigation
