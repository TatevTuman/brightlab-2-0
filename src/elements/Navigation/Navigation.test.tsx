import React from 'react'
import { cleanup, render, waitFor } from '@testing-library/react'
import Navigation from './Navigation'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Navigation', () => {
  const navigation = new Array(4).fill(0).map((_, index) => ({
    path: `/page${index}`,
    label: `Page${index}`
  }))

  it('renders correctly', async () => {
    const { container } = render(<Navigation navigation={navigation} />)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
  })
})
