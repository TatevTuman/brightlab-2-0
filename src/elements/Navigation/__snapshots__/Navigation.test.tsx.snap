// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<div>
  <nav
    class="navigation"
    role="navigation"
  >
    <ul
      class="navigation-list"
    >
      <li
        class="navigation-list-item"
        data-active="false"
      >
        <div
          class="navigation-list-item-icon"
        />
        <div
          class="navigation-list-item-link"
        >
          Page0
        </div>
      </li>
      <li
        class="navigation-list-item"
        data-active="false"
      >
        <div
          class="navigation-list-item-icon"
        />
        <div
          class="navigation-list-item-link"
        >
          Page1
        </div>
      </li>
      <li
        class="navigation-list-item"
        data-active="false"
      >
        <div
          class="navigation-list-item-icon"
        />
        <div
          class="navigation-list-item-link"
        >
          Page2
        </div>
      </li>
      <li
        class="navigation-list-item"
        data-active="false"
      >
        <div
          class="navigation-list-item-icon"
        />
        <div
          class="navigation-list-item-link"
        >
          Page3
        </div>
      </li>
    </ul>
  </nav>
</div>
`;
